### PR TITLE
Add dashboard keyboard shortcuts

### PR DESCRIPTION
**Pull request by rohithreddykota**
Adds dashboard-scoped keyboard shortcuts and a discoverable help dialog.

- Press `?` to show or hide the keyboard-shortcuts menu.
- Press `/`, `,`, or `.` to open the filter, metric, or dimension picker.
- Lists the existing cell inspection, selection, chart navigation, zoom, and explain shortcuts in the same menu.
- Ignores shortcut keys while typing, when modifiers are held, or during repeated keydown events.

Tested with `npm test --workspace web-common -- src/features/dashboards/shortcuts/dashboard-shortcuts.spec.ts` (9 tests passing).

**Checklist:**
- [x] Covered by tests
- [ ] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [x] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [x] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

### CODE REVIEW

**Review by AdityaHegde — COMMENTED** [review]
Code wise this looks good. The only thing missing is localisation and text is hardcoded in english.

Feature wise, do we need more keyboard shortcuts? Feels wierd to be able to open the filter dropdown for example and arrow keys don't work.

File: web-common/src/features/dashboards/shortcuts/DashboardShortcuts.svelte [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  let open = false;
+
+  function handleKeydown(event: KeyboardEvent) {
+    const action = getDashboardShortcutAction(event);
+    if (!action || !performDashboardShortcut(action)) return;
+
+    event.preventDefault();
+    event.stopPropagation();
+  }
```
Comment (nishantmonu51) [inline web-common/src/features/dashboards/shortcuts/DashboardShortcuts.svelte:19]: `handleKeydown` never consults `open`, so the picker shortcuts stay live while the shortcuts dialog is up. Focus sits inside the body-portalled `Dialog.Content`, which is not an editable target, so `/`, `,` or `.` still resolves an action and `performDashboardShortcut` finds the trigger that is still mounted behind the overlay and clicks it. Both bits-ui triggers accept that synthetic click (`DropdownMenuTriggerState.onclick` only rejects `e.detail !== 0`, and `PopoverTriggerState.onclick` toggles on `button === 0`), so the dropdown opens behind the modal and the user is left with two overlapping surfaces to dismiss. The same holds for any other open modal, not just this one. Only `ToggleHelp` should be honoured while `open` is true.

File: web-common/src/features/dashboards/shortcuts/dashboard-shortcuts.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    action: DashboardShortcutAction.UndoZoom,
+    keys: ["⌘/Ctrl", "Z"],
+    description: "Undo chart zoom",
+  },
+  {
+    action: DashboardShortcutAction.Explain,
+    keys: ["E"],
+    description: "Explain the selected range",
+  },
+];
```
Comment (nishantmonu51) [inline web-common/src/features/dashboards/shortcuts/dashboard-shortcuts.ts:89]: @AdityaHegde already flagged the hardcoded copy; adding the specifics. All thirteen descriptions here, plus the dialog title, the description and the `aria-label` on the `sr-only` button in `DashboardShortcuts.svelte:27,36-39`, are literal English, while `AddExpressionFilterButton.svelte` already reads its copy from paraglide two lines below the hunk this PR adds (`aria-label={m.dashboard_add_filter_button()}`). `scripts/i18n-guard.js` only scans files that import the messages namespace or match `MIGRATED_GLOBS`, and these new files do neither, so CI will not flag it and it becomes silent migration debt. The keys need to be added to both `web-common/src/lib/i18n/messages/en.json` and `es.json`.

**Review by nishantmonu51 — COMMENTED** [review]
A couple of smaller things that did not warrant their own thread:

- The help list is unconditional, but several rows are dead in common views: `,` has no target in time-dimension-detail or pivot (`MetricsTimeSeriesCharts.svelte:315-334` only renders the measure picker in the `{:else}` branch), `.` has no target outside the leaderboard branch (`Dashboard.svelte:271-283`), and `ChartInteractions.svelte:65` only handles `e` when `$explainEnabled`. Hiding rows whose target is absent would keep the dialog honest.
- Nine of the thirteen `DashboardShortcutAction` values exist only to key the help list, so `performDashboardShortcut(DashboardShortcutAction.Zoom)` type-checks but always returns `false`. A plain `id: string` on `DashboardShortcut` would express that split more honestly.
